### PR TITLE
fix(js): useKeyPress should define functions in useEffect

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Otherwise these functions are just redefined every time. Also changing the targetKey will not work.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: static/app/utils/useKeyPress.tsx [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
     window.addEventListener('keydown', downHandler);
     window.addEventListener('keyup', upHandler);
-    // Remove event listeners on cleanup
+
     return () => {
       window.removeEventListener('keydown', downHandler);
       window.removeEventListener('keyup', upHandler);
     };
-  }, []); // Empty array ensures that effect is only run on mount and unmount
+  }, [targetKey]);
```
Comment (priscilawebdev) [inline static/app/utils/useKeyPress.tsx:28]: now always when `targetKey` changes, we will be adding the same `EventListener` over and over... do we want this?

Comment (dashed) [reply static/app/utils/useKeyPress.tsx:28]: Only whenever `targetKey` changes. We've only been using `useKeyPress` with `targetKey` hardcoded to a string.

Comment (EvanPurkhiser) [reply static/app/utils/useKeyPress.tsx:28]: Yeah, it will clean up the old event listeners too. This is a good change

Comment (priscilawebdev) [reply static/app/utils/useKeyPress.tsx:28]: in this case it's fine :)

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**